ADD: 레벨 전환 (LoadStreamLevel), 스폰 Z축 수정

Jo-sm94 opened the pull request:
레벨 전환 (LoadStreamLevel)
 - default Level에서 Streaming Level(웨이브로 사용할 레벨) 추가하여 레벨 전환
 - 이전에 사용한 웨이브 레벨 언로드 후 다음 웨이브 레벨 로드하는 방향으로 진행

스폰 Z축 수정(바닥에서 스폰됨)
 - 라인트레이스 사용하여 지면에 충돌된 위치에서 스폰되게끔 수정

Jo-sm94 committed:
ADD: 레벨 전환 (LoadStreamLevel), 스폰 Z축 수정

레벨 전환 (LoadStreamLevel)
 - default Level에서 Streaming Level(웨이브로 사용할 레벨) 추가하여 레벨 전환
 - 이전에 사용한 웨이브 레벨 언로드 후 다음 웨이브 레벨 로드하는 방향으로 진행

스폰 Z축 수정(바닥에서 스폰됨)
 - 라인트레이스 사용하여 지면에 충돌된 위치에서 스폰되게끔 수정

Bathboomsoso commented on Source/Shooter/Private/GameModes/ShooterGameInstance.cpp:
```diff
@@ -38,22 +39,54 @@ void UShooterGameInstance::Init()
+
+	UGameplayStatics::LoadStreamLevel(
+		this,														//함수에 호출될 월드 this나 GetWolrd
+		LevelToLoad,												//로드할 레벨 이름
```
조아용!!! 비동기 로딩 나이서~ 이전 레벨 언로드도 자원 관리상으로 아주 좋은것 같습니다.

hwarang98 replied:
요거 아마 동기로드 일겁니다 ㅋㅋㅋㅋ

Bathboomsoso commented on Source/Shooter/Private/GameModes/Spawner.cpp:
```diff
@@ -55,14 +56,43 @@ int32 ASpawner::SpawnEnemies(const TArray<TSubclassOf<class ACharacter>>& EnemyT
 	FVector BoxExtent = SpawningBox->GetScaledBoxExtent();
 	FVector BoxOrigin = SpawningBox->GetComponentLocation();
 
-	return BoxOrigin + FVector(
```
제가 8번 과제 했을때 스폰 위치 탐색 재시도 로직 한번 추가했었는데, 스폰이 콜리전 충돌로 실패시 다른 좌표에서 스폰 시도해서 30마리면 딱 30마리 스폰되게 로직 한번 구현하는것도 좋아보입니다.
적 캐릭터가 겹쳐져서 스폰되는 현상을 아마 해결할 수 있을거에요
SpawnActor 함수에 콜리전 충돌관련 옵션 있는걸로 아는데 한번 확인해보시면 될거같습니다.!!

Bathboomsoso commented:
주말인데도 고생많으셨습니다. !! ㅎ

hwarang98 commented on Source/Shooter/Private/GameModes/ShooterGameInstance.cpp:
```diff
@@ -38,22 +39,54 @@ void UShooterGameInstance::Init()
+	UWorld* World = GetWorld();
+	if (!World) return;
+
+	AShooterBaseGameMode* GM = GetWorld()->GetAuthGameMode<AShooterBaseGameMode>();
```
상단에 World변수를 사용하는게 더 깔끔해보입니다!

Jo-sm94 committed:
레벨 전환 (LoadStreamLevel), 스폰 Z축 수정 #2

레벨 전환 (LoadStreamLevel)
 - default Level에서 Streaming Level(웨이브로 사용할 레벨) 추가하여 레벨 전환
 - 이전에 사용한 웨이브 레벨 언로드 후 다음 웨이브 레벨 로드하는 방향으로 진행

스폰 Z축 수정(바닥에서 스폰됨)
 - 라인트레이스 사용하여 지면에 충돌된 위치에서 스폰되게끔 수정
 - 스폰 충돌 시 스폰위치 설정하여 생성되도록 수정  (#2 추가 부분)
